pkg/domain/domain.go

```diff
@@ -473,14 +473,34 @@ func (do *Domain) tryLoadSchemaDiffs(m *meta.Meta, usedVersion, newVersion int64
 	phyTblIDs := make([]int64, 0, len(diffs))
 	actions := make([]uint64, 0, len(diffs))
 	diffTypes := make([]string, 0, len(diffs))
-	for _, diff := range diffs {
+	for i, diff := range diffs {
```

Comment:
Why not wrap the content of the old for loop as a function, and call them len(diffs)-1 times?

Like:

```
for _, diff := range diffs {
    builder := NewBuilder()
    builder.ApplyDiff()
    builder.Build()   // or BuildWithoutUpdateBundle
}
```

Reply:
I think it would be more complicated to do that.

Reply:
I don't need multiple builders here, maybe it's better to extract a `infoschema.clone` function to use in here.